server/src/main/java/org/apache/druid/server/StatusResource.java

```diff
@@ -92,15 +93,15 @@ private Map<String, String> filterHiddenProperties(
       Map<String, String> allProperties
   )
   {
-    return Maps.filterEntries(
-        allProperties,
-        (entry) -> hiddenProperties
-            .stream()
-            .anyMatch(
-                hiddenPropertyElement ->
-                    !StringUtils.toLowerCase(entry.getKey()).contains(StringUtils.toLowerCase(hiddenPropertyElement))
-            )
+    Map<String, String> propertyCopy = new HashMap<>(allProperties);
+    allProperties.keySet().forEach(
+        (key) -> {
+          if (hiddenProperties.stream().anyMatch((hiddenProperty) -> StringUtils.toLowerCase(key).contains(StringUtils.toLowerCase(hiddenProperty)))) {
```

Comment:
nit: So if we have set of hiddenProps as =["pwd"] and in the allProperties we have {"foopwd"->"secret1" , "pwd"->"secret", "foo"->"bar"  } the output would be {"foo"->"bar"} . 
In that case, we should update a JavaDoc to reflect this